content/openapi/types-and-parameters.md

```diff
@@ -0,0 +1,290 @@
+### Types and parameters
+
+The `SwaggerModule` searches for all `@Body()`, `@Query()`, and `@Param()` decorators in route handlers to generate the API document. It also creates corresponding model definitions by taking advantage of reflection. Consider the following code:
+
+```typescript
+@Post()
+async create(@Body() createCatDto: CreateCatDto) {
+  this.catsService.create(createCatDto);
+}
+```
+
+> info **Hint** To explicitly set the body definition use the `@ApiBody()` decorator (`@nestjs/swagger` package).
+
+Based on the `CreateCatDto`, the module definition will be created:
+
+<figure><img src="/assets/swagger-dto.png" /></figure>
+
+As you can see, the definition is empty although the class has a few declared properties. In order to make the class properties visible to the `SwaggerModule`, we have to either annotate them with the `@ApiProperty()` decorator or use a CLI plugin (read more in the **Plugin** section) which will do it automatically:
+
+```typescript
+import { ApiProperty } from '@nestjs/swagger';
+
+export class CreateCatDto {
+  @ApiProperty()
+  name: string;
+
+  @ApiProperty()
+  age: number;
+
+  @ApiProperty()
+  breed: string;
+}
+```
+
+> info **Hint** Consider using the Swagger plugin (see [Plugin](/recipes/swagger#plugin) section) which will automatically do it for you.
+
+Let's open the browser and verify the generated `CreateCatDto` model:
+
+<figure><img src="/assets/swagger-dto2.png" /></figure>
+
+In addition, the `@ApiProperty()` decorator allows setting various [Schema Object](https://swagger.io/specification/#schemaObject) properties:
+
+```typescript
+@ApiProperty({
+  description: 'The age of a cat',
+  minimum: 1,
+  default: 1,
+})
+age: number;
+```
+
+> info **Hint** Instead of explicitly typing the `{{"@ApiProperty({ required: false })"}}` you can use `@ApiPropertyOptional()` short-hand decorator.
```

Comment:
```suggestion
> info **Hint** Instead of explicitly typing the `{{"@ApiProperty({ required: false })"}}` you can use the `@ApiPropertyOptional()` short-hand decorator.
```